### DISCUSSION TITLE
Graphical User Interface

### DISCUSSION THREAD

**Discussion by hamishbeath** [CLOVER-energy/CLOVER#200] [IAA]
This would be part of the 'expanding user base' section of the IAA grant and would allow wider use from a non-technical background. The aim would be to implement a very simple GUI that wraps around existing code and facilitates easier input/output management.

**Comment by BenWinchester** [comment]
## General thoughts
Having a graphical user interface (GUI) seems like an easy win in terms of accessibility, and, so, I reckon this is something that we should definitley consider as part of the IAA grant. The GUI should be easily navigable by a user who hasn't got much experience (if any) of using Python-based tools, so, ideally, it would be runnable without using the command-line interface (CLI). It also might be a good idea to beta test ideas, as well as sketches and flows, with the wider existing userbase to see whether any designs that we come up with are suitable and work well.

## Navigation
Currently, CLOVER is split into three streams:
1. **Running _without_ either a simulation or an optimisation** - CLOVER can be run from the CLI in such a way that its only job is to generate profiles for device utilisation, usership etc., as well as downloading solar resource data from the `renewables.ninja` API;
2. **Running a simulation** - This simply assesses the performance of a system over its lifetime,
   a. _without_ any analysis or generation of default plots being done,
   b. _with_ generation of default plots as well as an appraisal of the system;
3. **Running an optimisation** - This runs multiple simulations to search the space of available system capacities for the optimum system.

The first stream, _running without either a simulation or an optimisation_, is functionality that likely doesn't need to be covered within the GUI as this can be done automatically and is faff that the user could do without. The other two, _simulations_ and _optimisations_, should be covered by separate streams, but with the same interface for selecting components which are present as well as determining which types of components are selected for.

**Note:** currently, downloading data from `renewables.ninja` uses a specific tilt angle. This functionality may need to be changed as multiple PV arrays can be selected or deselected more easily using the GUI.

## Sketches
A sketch on the GUI is given below.
[Thoughts on a GUI.pdf](https://github.com/CLOVER-energy/CLOVER/files/11676432/Thoughts.on.a.GUI.pdf)

## ⬆️ Promotion to an issue ⬆️ 
This is likely something that we will consider as part of the IAA grant, so is being promoted to a new issue within the repository.
